client/state/global-sidebar/selectors.ts

```diff
@@ -70,21 +82,37 @@ export const getShouldShowCollapsedGlobalSidebar = (
 	const isSitesDashboard = sectionGroup === 'sites-dashboard';
 	const isSiteDashboard = getShouldShowSiteDashboard( state, siteId, sectionGroup, sectionName );
 
-	// A site is just clicked and the global sidebar is in collapsing animation.
-	const isSiteJustSelectedFromSitesDashboard =
+	if ( collapsedDataHelper.sectionGroup !== sectionGroup ) {
+		if ( isSitesDashboard ) {
+			// Set or refresh the initial value when loading into the dashboard.
+			collapsedDataHelper.selectedSiteId = siteId;
+		} else {
+			// Clear this once we are off the sites dashboard.
+			collapsedDataHelper.shouldShowForAnimation = false;
+		}
+		// Keep track of section group to evaluate things when this changes.
+		collapsedDataHelper.sectionGroup = sectionGroup;
+	}
+
+	// When selected site changes on the dashboard, show for animation.
+	if (
 		isSitesDashboard &&
 		!! siteId &&
-		isInRoute( state, [
-			'/sites', // started collapsing when still in sites dashboard
-			...Object.values( SITE_DASHBOARD_ROUTES ), // has just stopped collapsing when in one of the paths in site dashboard
-		] );
+		collapsedDataHelper.selectedSiteId !== siteId &&
+		! collapsedDataHelper.shouldShowForAnimation
+	) {
+		collapsedDataHelper.shouldShowForAnimation = true;
+		collapsedDataHelper.selectedSiteId = siteId;
+	}
 
 	const isPluginsScheduledUpdatesEditMode =
 		isScheduledUpdatesMultisiteCreateRoute( state ) ||
 		isScheduledUpdatesMultisiteEditRoute( state );
 
 	return (
-		isSiteJustSelectedFromSitesDashboard || isSiteDashboard || isPluginsScheduledUpdatesEditMode
+		collapsedDataHelper.shouldShowForAnimation ||
```

Comment:
This updated logic for when to show the collapsed sidebar strictly to trigger the animation early seems to work pretty well across all the cases I test (except one noted below).

There is 1 edge case where this fails to happen: **if we have a site selected and then select that same site** in the dashboard, this animation logic will not trigger correctly making it look a bit jumpy. This only seems to happen when we go from a site context, to the dashboard, and then select the same site:

![jumpy-anim](https://github.com/user-attachments/assets/743a2db2-d3fc-472a-b3c6-1931727cf4b9)

Notice the sidebar animation triggers a little later than it normally would in this case.

I tried a couple logical solutions here without success. I think what we need to do is carve out a small store in redux state to store some of this, so that when we select an item through the dashboard component it sets a similar just selected value (regardless if it is already the currently selected site) - and have that same thing cleared whenever the section group changes. I think that would be the way to make this more robust going forward and resolve that edge case noted above.

Reply:
Let's create a separate follow up issue for this rather than hold up this PR for it.

Reply:
Created a followup issue: https://github.com/Automattic/dotcom-forge/issues/8325